docs/Typechecking.md

```diff
@@ -1,47 +1,97 @@
 # Type Checking With Sorbet
```

Comment:
A definite improvement here, nice work 👏🏻. Would be good to have `brew typecheck` output a link to these docs on any failures (and `brew style`, if possible, if it's Sorbet-related?)